# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [19]:
import pandas as pd
import numpy as np
from scipy import stats
import math

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='w'].call)

235.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

<div class="span5 alert alert-success">
<p>Your answers to Q1 and Q2 here</p>
</div>

**What test is appropriate for this problem, Does CLT Apply ?**

We should clarify the population of "W" and "B" to establish the base case (50% W, 50% B) and then compare that against the callback numbers. We would expect a roughly  
similar call back ratio (assuming the resumes meeting the job requirements).

CLT (Central Limit Theorem) does apply here. There are 4,870 observations and the mean response for job call backs should be roughtly the same for both "W" and "B" 

** What are the null and alternative hypothesis**

The null hypothesis would be that their is not racial discrimination based on the name of the job applicant. The job response rate for both "W" and "B" applicants should  
be the same.

The alternative hypothesis is the the response rate is NOT the same for the "W" and "B" job applicants.


**Question 3 Calculations Below**


In [13]:
w = data[data.race=='w']
b = data[data.race=='b']
w_num = len(w)
b_num = len(b)
print(f' The number of observations for "W" is {w_num}')
print(f' The number of observations for "B" is {b_num}')

 The number of observations for "W" is 2435
 The number of observations for "B" is 2435


The measure of sucesss is getting a call-back (call = 1). Call = 0 is a failure (i.e. - no callback)

In [17]:
w_callback_num = len(w[w.call==1])
b_callback_num = len(b[b.call==1])

print(f' The number of call backs for "W" is {w_callback_num}')
print(f' The number of call backs for "B" is {b_callback_num}')

w_callback_ratio = w_callback / w_num
b_callback_ratio = b_callback / b_num

print(f' The percent of call backs for "W" is {w_callback_ratio * 100:.02f}%')
print(f' The percent of call backs for "B" is {b_callback_ratio * 100:.02f}%')



 The number of call backs for "W" is 235
 The number of call backs for "B" is 157
 The percent of call backs for "W" is 9.65%
 The percent of call backs for "B" is 6.45%


The sample size (4870) supports using the CLT and using the z-statistic for a confidence interval on the sample statistic. For a 95% confidence interval, the z-value is 1.96

In [28]:
z = 1.96
# margin_error = z * math.sqrt( ( p_w*(1-p_w) / num_b) + (p_b*(1-p_b)/num_w) )
margin_error = z * math.sqrt( ( w_callback_ratio*(1- w_callback_ratio) / b_num) + (b_callback_ratio*(1-b_callback_ratio)/w_num) )

print(f'Margin of error = {margin_error:04f}')

print(f'The confidence interval is: {w_callback_ratio - b_callback_ratio-z*margin_error:04f} to {w_callback_ratio - b_callback_ratio+z*margin_error:04f}')

Margin of error = 0.015255
 The confidence interval is: 0.002132 to 0.061933


Calculate the p-value

In [29]:
from statsmodels.stats.proportion import proportions_ztest as pz
pz(np.array([b_callback_num,w_callback_num]),np.array([b_num,w_num]),value=0)

(-4.108412152434346, 3.983886837585077e-05)

<div class="span5 alert alert-success">
<p> Your answers to Q4 and Q5 here </p>
</div>

**Q4:**
The second number above (3.983e-05) is the p-value. It is significantly below 0.05 therefore we can reject the null hypothesis

The null hypothesis that the job call backs for "W" and "B" race sounding names would be equal is rejected by the low p-value. While the "W" call backs is 9.65% and the "B" callbacks is 6.45%,   
a difference of 34%, the low p-value confirms this

**Q5:**
The race sounding names are one measure of a call back but their could be other factors (experience, education, location, etc.). Additional measures should be taken to make the resumes as 
equal as possible across multiple dimensions. Another approach would be to actually make the "B" resumes even more qualified than the "W" to see if that makes a difference.